packages/core/src/RenderingEngine/VolumeViewport.ts

```diff
@@ -88,40 +88,47 @@ class VolumeViewport extends Viewport implements IVolumeViewport {
   }
 
   private initializeVolumeNewImageEventDispatcher(): void {
-    const volumeNewImageHandler = (cameraEvent) => {
+    function volumeNewImageHandler(cameraEvent) {
       const viewportImageData = this.getImageData();
 
       if (!viewportImageData) {
         return;
       }
 
       volumeNewImageEventDispatcher(cameraEvent);
-    };
+    }
+
+    function volumeNewImageCleanUp(evt) {
+      eventTarget.removeEventListener(
```

Comment:
Doesn't this have to come after the if block on line 109-111? Right now _another_ viewport being disabled will cause this handler to be removed from this viewport.